### PR TITLE
VEOSS-595 | Added validation for correction amount

### PR DESCRIPTION
**Pull request by owen-opencast**
Also including content from [VEOSS-666.](https://jira.tools.tax.service.gov.uk/browse/VEOSS-666)

### CODE REVIEW

File: test/controllers/corrections/CountryVatCorrectionControllerSpec.scala [hunk 1]
```diff
@@ -166,6 +176,73 @@ class CountryVatCorrectionControllerSpec extends SpecBase with MockitoSugar {
...
+      )
+
+      val previousCorrection = CorrectionToCountry(
+        selectedCountry,
+        BigDecimal(100.0)
+      )
+
+      when(mockVatReturnConnector.get(any())(any())) thenReturn Future.successful(Right(previousVatReturn))
+      when(mockCorrectionService.getCorrectionsForPeriod(any())(any(), any())) thenReturn Future.successful(List(previousCorrection))
+//      when(mockService.getLatestVatAmountForPeriodAndCountry(any(), any())(any(), any())) thenReturn Future.successful(validAnswer)
```
Comment (darbyo) [inline test/controllers/corrections/CountryVatCorrectionControllerSpec.scala:212]: Do we need these mocks? If not can delete the comments

**Review by darbyo — CHANGES_REQUESTED** [review]

File: app/controllers/corrections/CountryVatCorrectionController.scala [hunk 2]
```diff
@@ -48,8 +48,7 @@ class CountryVatCorrectionController @Inject()(
           for{
             vatOwedToCountryOnPrevReturn <- service.getLatestVatAmountForPeriodAndCountry(country, correctionPeriod)
           }yield{
-            val form = formProvider(country.name)
-
+            val form = formProvider(country.name, -vatOwedToCountryOnPrevReturn)
```
[the remark this answers is not in the document]
Comment (owen-opencast) [reply app/controllers/corrections/CountryVatCorrectionController.scala:51]: We could, we'd need to add a custom validator to checking of that form. We'd also still need to do -vatOwedToCountryOnPrevReturn in the form provider, to get the error message in the format specified (e.g. "The correction value must be more than £-300" - the inverse of the current VAT owed). Can do it if you want to?

[the remark this answers is not in the document]
Comment (darbyo) [reply app/controllers/corrections/CountryVatCorrectionController.scala:51]: I think we should go for the more readable solution. But will leave the decision up to you guys, I think this way isnt totally obvious what is happening until you dig down inside it, however it is correct. If yous dont want to make that change, can we at least pull '-vatOwedToCountryOnPrevReturn' out to a variable named such as 'minimumAmount' or something nicely descriptive

[the remark this answers is not in the document]
Comment (owen-opencast) [reply app/controllers/corrections/CountryVatCorrectionController.scala]: I've added pulled it out to a new variable, I think it's clearer now!

**Review by darbyo — APPROVED** [review]

**Pull request merged**